### PR TITLE
[Bindings.Sofa.Helper] Add MessageHandler context manager class

### PR DESCRIPTION
**Pull request by marques-bruno**
This PR lets you write MessageHandlers in Python.
It also lets you instantiate them in a RAII way by taking advantage of Python3's context managers.

All you have to do is implement the process function in a MessageHandler-inherited python class (see diff in Message.py)
@damienmarchal @jnbrunet

### CODE REVIEW

File: bindings/Sofa/src/SofaPython3/Sofa/Helper/Binding_MessageHandler.cpp [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+        f.def(py::init([]()
+        {
+            return new MessageHandler_Trampoline();
+        }));
+
+        f.def("process", &PyMessageHandler::process);
+        f.def("__enter__", [](PyMessageHandler* self) -> PyMessageHandler*
+        {
+            PythonEnvironment::gil acquire {"MessageHandler"};
```
Comment (damienmarchal) [inline bindings/Sofa/src/SofaPython3/Sofa/Helper/Binding_MessageHandler.cpp:125]: Why a gil ? Isn't this a pure c++ code ?

File: bindings/Sofa/src/SofaPython3/Sofa/Simulation/Submodule_Simulation.cpp [hunk 2]
```diff
@@ -40,6 +40,11 @@ using namespace pybind11::literals;
 using sofa::simulation::Simulation;
 
 #include <sofa/core/visual/VisualParams.h>
+#include <sofa/helper/logging/MessageDispatcher.h>
+#include <sofa/helper/logging/ConsoleMessageHandler.h>
+#include <sofa/core/logging/PerComponentLoggingMessageHandler.h>
+using sofa::helper::logging::MessageDispatcher;
+using sofa::helper::logging::MainPerComponentLoggingMessageHandler;
```
Comment (damienmarchal) [inline bindings/Sofa/src/SofaPython3/Sofa/Simulation/Submodule_Simulation.cpp:47]: These includes are useless now isn't it ?

File: bindings/SofaRuntime/src/SofaPython3/SofaRuntime/Module_SofaRuntime.cpp [hunk 3]
```diff
@@ -130,6 +137,12 @@ PYBIND11_MODULE(SofaRuntime, m) {
         return simpleapi::importPlugin(name);
     }, "import a sofa plugin into the current environment");
 
+    m.def("init", []() {
+        MessageDispatcher::clearHandlers();
+        MessageDispatcher::addHandler(&MainConsoleMessageHandler::getInstance());
```
Comment (damienmarchal) [inline bindings/SofaRuntime/src/SofaPython3/SofaRuntime/Module_SofaRuntime.cpp:142]: Il n'y a pas de singletont pour ConsoleMessageHandler.

File: bindings/Sofa/src/SofaPython3/Sofa/Simulation/CMakeLists.txt [hunk 4]
```diff
@@ -18,5 +18,5 @@ SP3_add_python_module(
         MODULE_NAME  Simulation
         SOURCES      ${SOURCE_FILES}
         HEADERS      ${HEADER_FILES}
-        DEPENDS      SofaCore SofaSimulationCore SofaSimulationGraph SofaBaseVisual
-)
\ No newline at end of file
+        DEPENDS      SofaHelper SofaCore SofaSimulationCore SofaSimulationGraph SofaBaseVisual
```
Comment (damienmarchal) [inline bindings/Sofa/src/SofaPython3/Sofa/Simulation/CMakeLists.txt:21]: Not needed anymore ?

File: bindings/SofaRuntime/src/SofaPython3/SofaRuntime/Module_SofaRuntime.cpp (hunk 3 above)
Comment (marques-bruno) [reply bindings/SofaRuntime/src/SofaPython3/SofaRuntime/Module_SofaRuntime.cpp:142]: Désolé oublié de référence la PR SOFA associée : https://github.com/sofa-framework/sofa/pull/1434

**Comment by jnbrunet** [comment]
Ok for me, and tests have passed :+1:

**Pull request merged**